annotations/tasks.py

```diff
@@ -278,3 +293,69 @@ def handle_file_upload(request, form):
     if file_content != None:
         tokenized_content = tokenize(file_content)
         return save_text_instance(tokenized_content, text_title, date_created, is_public, user, uri)
+
+
+
+
+
+
+@shared_task
+def process_import_task(user, project_id ,file, part_of_id, action, name):
+    with open(file, 'r') as f:
+        csv_file = csv.reader(f)
+        # there isn't a good way to reset the iterator so we tee them up
+        reader1, reader2 = tee(csv_file)
+        reader1.next()
+        reader2.next()
+        total_count = len(list(reader2))
+        count = 0
+        task = ImportTasks.objects.create(
+            user = user,
+            file_name = name,
+            total_rows = int(total_count)
+        )
+        for row in reader1:
+            # Store url so we can skip making the request to fetch the json if the urls are the same
+            stored_url = ''
+            try:
+                parent = Text.objects.get(uri=row[3])
+                text = Text.objects.get(part_of_id=parent.id)
+                occur = text
+            except:
+                # Only make the request and get the json if the url has changed.
+                url = settings.AMPHORA_RESOURCE_URL + row[3] + "&format=json"
+                if stored_url != url:
+                    stored_url = url
+                    text_request = requests.get(url, headers=auth.jars_github_auth(user))
+                    print(auth.jars_github_auth(user))
+                    text_json = text_request.json()
+                    #find the first text in content
+                    for content in text_json['content']:
+                        if content['content_resource']['content_type'] == 'text/plain':
+                            text_content = content['content_resource']['id']
+                            # Save time and stop the iteration
+                            break
```

Comment:
downloading a text from amphora should be in its own method (preferably in its own package)